File: backend/src/main/java/com/wgzhao/addax/admin/model/SysItemPK.java
package com.wgzhao.addax.admin.model;

import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;
import java.util.Objects;

@Getter
@Setter
public class SysItemPK
        implements Serializable
{
    private Integer dictCode;
    private String itemKey;

    public SysItemPK()
    {
    }

    public SysItemPK(Integer dictCode, String itemKey)
    {
        this.dictCode = dictCode;
        this.itemKey = itemKey;
    }

    @Override
    public boolean equals(Object o)
    {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        SysItemPK that = (SysItemPK) o;
        return dictCode.equals(that.dictCode) && itemKey.equals(that.itemKey);
    }

    @Override
    public int hashCode()
    {
        return Objects.hash(dictCode, itemKey);
    }
}

File: src/main/kotlin/com/wgzhao/addax/admin/model/SysItemPK.kt
package com.wgzhao.addax.admin.model

import lombok.Getter
import lombok.Setter
import java.io.Serializable
import java.util.*

@Getter
@Setter
class SysItemPK

    : Serializable {
    private var dictCode: Int? = null
    private var itemKey: String? = null

    constructor()

    constructor(dictCode: Int, itemKey: String) {
        this.dictCode = dictCode
        this.itemKey = itemKey
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o == null || javaClass != o.javaClass) {
            return false
        }
        val that = o as SysItemPK
        return dictCode == that.dictCode && itemKey == that.itemKey
    }

    override fun hashCode(): Int {
        return Objects.hash(dictCode, itemKey)
    }
}
